go/vt/vttablet/onlineddl/executor.go

```diff
@@ -666,27 +691,41 @@ func (e *Executor) doesConnectionInfoMatch(ctx context.Context, connID int64, su
 	return len(rs.Rows) == 1, nil
 }
 
-// validateTableForAlterAction checks whether a table is good to undergo a ALTER operation. It returns detailed error if not.
-func (e *Executor) validateTableForAlterAction(ctx context.Context, onlineDDL *schema.OnlineDDL) (err error) {
-	// Validate table does not participate in foreign key relationship:
+// tableParticipatesInForeignKeyRelationship checks if a given table is either a parent on a child in at least one foreign key constraint
+func (e *Executor) tableParticipatesInForeignKeyRelationship(ctx context.Context, schema string, table string) (bool, error) {
 	for _, fkQuery := range []string{selSelectCountFKParentConstraints, selSelectCountFKChildConstraints} {
 		query, err := sqlparser.ParseAndBind(fkQuery,
-			sqltypes.StringBindVariable(onlineDDL.Schema),
-			sqltypes.StringBindVariable(onlineDDL.Table),
+			sqltypes.StringBindVariable(schema),
+			sqltypes.StringBindVariable(table),
 		)
 		if err != nil {
-			return err
+			return false, err
 		}
 		r, err := e.execQuery(ctx, query)
 		if err != nil {
-			return err
+			return false, err
 		}
 		row := r.Named().Row()
 		if row == nil {
-			return vterrors.Errorf(vtrpcpb.Code_UNKNOWN, "unexpected result from INFORMATION_SCHEMA.KEY_COLUMN_USAGE query: %s", query)
+			return false, vterrors.Errorf(vtrpcpb.Code_UNKNOWN, "unexpected result from INFORMATION_SCHEMA.KEY_COLUMN_USAGE query: %s", query)
```

Comment:
> r == nil // shouldn't happen with a nil error

Correct, won't happen; we use this pattern throughout the vitess codebase. If `err == nil` we don't check whether `r == nil`. If this were a possibility `vitess` would break all over the place.

> len(r.Rows) != 1

The function `Row()` returns `nil` when `len(r.Rows) != 1`, and so the check `if row == nil` is exactly what we need.

Reply:
Ah, nice. I didn't realize that.